gateway/utils/syncman/store_sc.go

```diff
@@ -0,0 +1,195 @@
+					logrus.Errorf("error while watching projects in kube store unable to unmarshal data - %v", err)
+					continue
+				}
+				projectMap[key] = v
+			}
+			cb(s.getProjects(projectMap))
+
+		case watch.Deleted:
+			configMap := ee.Object.(*v1.ConfigMap)
+			for key := range configMap.Data {
+				delete(projectMap, key)
+			}
+			cb(s.getProjects(projectMap))
+		}
+	}
+	return nil
+}
+
+func (s *KubeStore) WatchServices(cb func(scServices)) error {
+	services := scServices{}
+	labels := fmt.Sprintf("app=%s,clusterId=%s", "service", s.clusterID) // todo verify this
+	watcher, err := s.kube.CoreV1().Pods(fmt.Sprintf("sc/instaces/%s", s.clusterID)).Watch(v12.ListOptions{Watch: true, LabelSelector: labels})
+	if err != nil {
+		return err
+	}
+	defer watcher.Stop()
+	for ee := range watcher.ResultChan() {
+		switch ee.Type {
+		case watch.Added:
+			pod := ee.Object.(*v1.Pod)
+			id, ok := pod.Annotations["id"] // "verify all off these
+			if !ok {
+				logrus.Errorf("error occurred watching services in kube store unable to find id in pod annotations while add event occurred")
+				break
+			}
+			addr, ok := pod.Annotations["addr"]
+			if !ok {
+				logrus.Errorf("error occurred watching services in kube store unable to find addr in pod annotations while add event occurred")
+				break
+			}
+			services = append(services, &service{id: id, addr: addr})
+			sort.Stable(services)
+			cb(services)
+
+		case watch.Modified:
+			pod := ee.Object.(*v1.Pod)
+			id, ok := pod.Annotations["id"]
+			if !ok {
+				logrus.Errorf("error occurred watching services in kube store unable to find id in pod annotations while modified event occurred")
+				break
+			}
+			addr, ok := pod.Annotations["addr"]
+			if !ok {
+				logrus.Errorf("error occurred watching services in kube store unable to find addr in pod annotations while modified event occurred")
+				break
+			}
+			count := 0
+			for _, service := range services {
+				if service.id == id {
+					count++
+					service.addr = addr
+					sort.Stable(services)
+					cb(services)
+					break
+				}
+			}
+			// if doesn't exit add to services
+			if count == 0 {
+				cb(append(services, &service{id: id, addr: addr}))
+			}
+
+		case watch.Deleted:
+			pod := ee.Object.(*v1.Pod)
+			id, ok := pod.Annotations["id"]
+			if !ok {
+				logrus.Errorf("error occurred watching services in kube store unable to find id in pod annotations while delete event occurred")
+				break
+			}
+			for index, service := range services {
+				if service.id == id {
+					services[index] = services[len(services)-1]
+					services = services[:len(services)-1]
+					sort.Stable(services)
+					cb(services)
+					break
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (s *KubeStore) SetProject(ctx context.Context, project *config.Project) error {
+	// TODO WHO WILL CREATE THIS CONFIG MAP
+	configMap, err := s.kube.CoreV1().ConfigMaps(fmt.Sprintf("sc/projects/%s", s.clusterID)).Get("", v12.GetOptions{})
```

Comment:
If the config map does not exist, means you need to create one instead. Check if it doesn't exists using kubeerrors. I've used it in the runner.

Reply:
Also set the labels of the config map. `projectId` and `clusterId`

Reply:
.Get() will have a string which is clusterId-projectId